server/service/appconfig.go

```diff
@@ -109,7 +106,22 @@ func (svc *Service) AppConfig(ctx context.Context) (*fleet.AppConfig, error) {
 		}
 	}
 
-	return svc.ds.AppConfig(ctx)
+	ac, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if ac.SMTPSettings.SMTPPassword != "" {
+		ac.SMTPSettings.SMTPPassword = "********"
+	}
+
+	var jiraIntegrations []*fleet.JiraIntegration
+	for _, jiraIntegration := range ac.Integrations.Jira {
+		jiraIntegration.APIToken = "********"
+		jiraIntegrations = append(jiraIntegrations, jiraIntegration)
+	}
+	ac.Integrations.Jira = jiraIntegrations
+	return ac, nil
```

Comment:
Nice, I like the idea of blocking them from the service method -- will make it harder for bugs to expose them in the future.